codegens/dart-http/lib/index.js

```diff
@@ -30,8 +30,19 @@ function parseUrlEncoded (body, indent, trim) {
  *
  * @param {Object} body Raw body data
  * @param {Boolean} trim indicates whether to trim string or not
+ * @param {String} contentType the content-type of request body
  */
-function parseRawBody (body, trim) {
+function parseRawBody (body, trim, contentType) {
+  if (contentType && (contentType === 'application/json' || contentType.match(/\+json$/))) {
+    try {
+      let jsonBody = JSON.parse(body);
+      return `request.body = json.encode(${JSON.stringify(jsonBody, null, 4)});`;
```

Comment:
The space will depend on the indentation right? @webholik

Reply:
@umeshp7 The problem is that if the indentation is using tabs, then we have no way to handle that with `JSON.stringify`. So I decided to use constant indentation for all cases.